pkg/test/test.msbuild

```diff
@@ -4,22 +4,31 @@
     <TestPackages Condition="'$(TestPackages)' != ''" Include="$(TestPackages)" />
   </ItemGroup>
 
+  <PropertyGroup>
+    <PackageTestProjectsDir Condition="'$(PackageTestProjectsDir)' == ''">$(MSBuildThisFileDirectory)..\projects</PackageTestProjectsDir>
+    <_projectProperties>SupportFilesDir=$(MSBuildThisFileDirectory)</_projectProperties>
+  </PropertyGroup>
+
   <Target Name="GetProjects">
     <ItemGroup>
-      <Project Condition="'@(TestPackages)' == ''" Include="projects\**\*.csproj" />
-      <Project Condition="'@(TestPackages)' != ''" Include="projects\%(TestPackages.Identity)\**\*.csproj" />
+      <Project Condition="'@(TestPackages)' == ''" Include="$(PackageTestProjectsDir)\**\*.csproj" />
+      <Project Condition="'@(TestPackages)' != ''" Include="$(PackageTestProjectsDir)\%(TestPackages.Identity)\**\*.csproj" />
     </ItemGroup>
   </Target>
 
   <Target Name="Restore"
           Inputs="@(Project)"
           Outputs="@(Project->'%(RootDir)%(Directory)obj\project.assets.json')"
           DependsOnTargets="GetProjects">
-    <MSBuild Projects="@(Project)" Targets="Restore" BuildInParallel="true" />
+    <MSBuild Projects="@(Project)" Targets="Restore" BuildInParallel="true" Properties="$(_projectProperties)" />
     <Touch Files="@(Project->'%(RootDir)%(Directory)obj\project.assets.json')" />
   </Target>
 
   <Target Name="Test" DependsOnTargets="GetProjects">
-    <MSBuild Projects="@(Project)" Targets="Test" BuildInParallel="true" />
+    <!-- We need to pass a dummy global property to force msbuild reevaluation after restoring the project to get implicit references added
+    and all generated props and targets imported in the test step. This is used to be able to call RestoreAndTest target when running in helix. -->
+    <MSBuild Projects="@(Project)" Targets="Test" BuildInParallel="true" Properties="$(_projectProperties);ForceMsbuildEvaluation=true" />
```

Comment:
Nifty, won't this cause binclashes though?

Reply:
From the runs that I've done I haven't seen binclashes happen, but it could potentially happen specially if we run ALL the tests in parallel using the RestoreAndTest target. I will try to get a binclash  and if so, we could think of a better workaround.